src/utils/pusher.tsx

```diff
@@ -4,62 +4,74 @@
  * This defines a Pusher client and channel connection as a vanilla Zustand store.
  */
 import Pusher, { Channel, PresenceChannel } from "pusher-js";
-import reactZustandCreate from "zustand";
 import vanillaCreate, { StoreApi } from "zustand/vanilla";
 
 const pusher_key = process.env.NEXT_PUBLIC_PUSHER_APP_KEY!;
+const pusher_server_host = process.env.NEXT_PUBLIC_PUSHER_SERVER_HOST!;
+const pusher_server_port = parseInt(process.env.NEXT_PUBLIC_PUSHER_SERVER_PORT!, 10);
+const pusher_server_tls = process.env.NEXT_PUBLIC_PUSHER_SERVER_TLS === 'true';
 
 interface PusherZustandStore {
   pusherClient: Pusher;
   channel: Channel;
   presenceChannel: PresenceChannel;
-
   members: { [key: string]: any };
 }
 
-// self-contained Pusher state machine that manages:
-// - Pusher Client
-// - Connected/subscribed channels
-// - "Current members" via presence
-const createPusherStore = (slug: string, publishPresence?: boolean) => {
-  const randomUserId = `random-user-id:${Math.random().toFixed(7)}`;
-  const pusherClient = new Pusher(pusher_key, {
-    authEndpoint: "/api/pusher/auth-channel",
-    auth: {
-      headers: { user_id: randomUserId },
-    },
-    cluster: "us3",
-  });
+const usePusherStore = (slug: string) => {
+  let pusherClient : Pusher;
+  if (Pusher.instances.length) {
```

Comment:
!!! very clever

Reply:
Had to dig into the internals of Pusher to figure that one out. Bashed my head on the keyboard for a minute before coming up with this solution 😂

Reply:
I might move back to the useState version I did on stream tbh it's the only "safe" way in modern react 😅